packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -1885,6 +1896,114 @@ export class ProjectService extends BaseService {
         return results;
     }
 
+    async streamResultsToLocalFile(
+        callback: (writer: (data: ResultRow) => void) => Promise<void>,
+    ): Promise<string> {
+        const downloadFileId = nanoid(); // Creates a new nanoid for the download file because the jobId is already exposed
+        const filePath = `/tmp/${downloadFileId}.jsonl`;
+        await this.downloadFileModel.createDownloadFile(
+            downloadFileId,
+            filePath,
+            DownloadFileType.JSONL,
+        );
+        const writeStream = fs.createWriteStream(filePath, {
+            encoding: 'utf8',
+        });
+
+        writeStream.on('error', (err) => {
+            this.logger.error('Error writing to file', err);
+            throw new UnexpectedServerError('Error writing to file');
+        });
+
+        const writer = (data: ResultRow) => {
+            writeStream.write(`${JSON.stringify(data)}\n`);
+        };
+
+        try {
+            await callback(writer);
+        } catch (err) {
+            this.logger.error('Error during streaming', err);
+            throw err;
+        } finally {
+            writeStream.end(() => {
+                this.logger.debug('File has been saved.');
+            });
+        }
+
+        return downloadFileId;
+    }
+
+    async streamSqlQueryIntoFile(
+        userUuid: string,
+        projectUuid: string,
+        sql: string,
+    ): Promise<string> {
+        const { organizationUuid } = await this.projectModel.getSummary(
+            projectUuid,
+        );
+
+        await this.analytics.track({
```

Comment:
I'm keeping the same analytics track, should we make a new one ? / add context ?

Reply:
We can add a new `usingStreaming: true` property in case we want to track usage during Beta.
btw, it's not a promise so we don't need the await.